### PR TITLE
network: allow users to forbid passthru MACVLAN from putting its link into promiscuous mode

### PR DESCRIPTION
**Pull request by tomty89**
While we haven't implemented a key for users to set MACVLAN/MACVTAP flags, we can at least allow them to make use of the Promiscuous= key of the corresponding link to set the nopromisc flag.

### CODE REVIEW

File: src/network/netdev/macvlan.c [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include "macvlan.h"
 #include "macvlan-util.h"
 #include "parse-util.h"
+#include "networkd-network.h"
```
Comment (poettering) [inline src/network/netdev/macvlan.c:8]: please keep the includes sorted alphabetically

File: src/network/netdev/macvlan.c [hunk 2]
```diff
@@ -52,6 +54,13 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_MACVLAN_MODE attribute: %m");
         }
 
+        // set the nopromisc flag if Promiscuous= of the link is explicitly set to false
```
Comment (poettering) [inline src/network/netdev/macvlan.c]: we use // comments as "fixme" style comments, and /* */ comments as explanatory comments. this is the latter, hence please use /**/.

See https://systemd.io/CODING_STYLE

**Review by poettering — CHANGES_REQUESTED** [review]
some superficial nits. i don't understand the effect of this, hence won't comment on this, i hope @yuwata knows more about this.

File: src/network/netdev/macvlan.c [hunk 3]
```diff
@@ -52,6 +54,13 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
...
+        if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU && !link->network->promiscuous) {
```
Comment (yuwata) [inline src/network/netdev/macvlan.c]: `link->network->promiscuous` is a tristate flag, and do not use implicit cast to boolean. So, use the following
```
if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU && link->network->promiscuous == 0) {
```
Or, maybe,
```
if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU && link->network->promiscuous >= 0) {
        r = sd_netlink_message_append_u16(req, IFLA_MACVLAN_FLAGS, link->network->promiscuous == 0 ? MACVLAN_FLAG_NOPROMISC : 0);
```
i.e. also enable promiscuous mode explicitly when `Promiscuous=yes`?

**Review by yuwata — CHANGES_REQUESTED** [review]
The basic concept looks ok.

File: src/network/netdev/macvlan.c (hunk 3 above)
Comment (tomty89) [reply src/network/netdev/macvlan.c]: I don't think we should / need to change the flags of the macvlan. AFAIK, the nopromisc flag only prevents the instruction from the macvlan side anyway.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**